account_voucher_supplier_invoice_number/voucher.py

```diff
@@ -19,47 +19,55 @@
 #
 ##############################################################################
 
-from openerp.osv import fields, orm
-from tools.translate import _
+from openerp import api, fields, models, _
 
-class voucher_line(orm.Model):
+class voucher_line(models.Model):
     _inherit = 'account.voucher.line'
     
-    def get_suppl_inv_num(self, cr, uid, move_line_id, context=None):
-        move_line = self.pool.get('account.move.line').browse(cr, uid, move_line_id, context)
+    def get_suppl_inv_num(self, move_line_id):
+        move_line = self.move_line_id.search([('id', '=', move_line_id)])
         return (move_line.invoice and move_line.invoice.supplier_invoice_number or '')
     
-    def _get_supplier_invoice_number(self, cr, uid, ids, name, args, context=None):
+
+    @api.multi
+    @api.depends(
+            'move_line_id',
+            'move_line_id.invoice',
+            'move_line_id.invoice.supplier_invoice_number'
+        )
+    def _get_supplier_invoice_number(self):
         res={}
-        for line in self.browse(cr, uid, ids, context):
+        for line in self:
             res[line.id] = ''
             if line.move_line_id:
-                res[line.id] = self.get_suppl_inv_num(cr, uid,
-                    line.move_line_id.id, context=context)
+                res[line.id] = self.get_suppl_inv_num(line.move_line_id.id)
+        print res
```

Comment:
New API you don't have to return a dictionary, but assign directly the value: `self.supplier_invoice_number = ...`